src/test/java/org/opensearch/flowframework/workflow/CreateIngestPipeline/CreateIngestPipelineStepTests.java

```diff
@@ -0,0 +1,143 @@
+
+            @Override
+            public Map<String, Object> getContent() {
+                return Map.ofEntries(
+                    Map.entry("id", "pipelineId"),
+                    Map.entry("description", "some description"),
+                    Map.entry("type", "text_embedding"),
+                    Map.entry("model_id", "model_id"),
+                    Map.entry("input_field_name", "inputField"),
+                    Map.entry("output_field_name", "outputField")
+                );
+            }
+        };
+
+        // Set output data to returned pipelineId
+        outpuData = new WorkflowData() {
+
+            @Override
+            public Map<String, Object> getContent() {
+                return Map.ofEntries(Map.entry("pipelineId", "pipelineId"));
+            }
+        };
+
+        client = mock(Client.class);
+        adminClient = mock(AdminClient.class);
+        clusterAdminClient = mock(ClusterAdminClient.class);
+
+        when(client.admin()).thenReturn(adminClient);
+        when(adminClient.cluster()).thenReturn(clusterAdminClient);
+    }
+
+    public void testCreateIngestPipelineStep() throws InterruptedException, ExecutionException {
+
+        CreateIngestPipelineStep createIngestPipelineStep = new CreateIngestPipelineStep(client);
+
+        ArgumentCaptor<ActionListener> actionListenerCaptor = ArgumentCaptor.forClass(ActionListener.class);
+        CompletableFuture<WorkflowData> future = createIngestPipelineStep.execute(List.of(inputData));
+
+        assertFalse(future.isDone());
+
+        // Mock put pipeline request execution and return true
+        verify(clusterAdminClient, times(1)).putPipeline(any(PutPipelineRequest.class), actionListenerCaptor.capture());
+        actionListenerCaptor.getValue().onResponse(new AcknowledgedResponse(true));
+
+        assertTrue(future.isDone() && !future.isCompletedExceptionally());
+        assertEquals(outpuData.getContent(), future.get().getContent());
+    }
+
+    public void testCreateIngestPipelineStepFailure() throws InterruptedException {
+
+        CreateIngestPipelineStep createIngestPipelineStep = new CreateIngestPipelineStep(client);
+
+        ArgumentCaptor<ActionListener> actionListenerCaptor = ArgumentCaptor.forClass(ActionListener.class);
+        CompletableFuture<WorkflowData> future = createIngestPipelineStep.execute(List.of(inputData));
+
+        assertFalse(future.isDone());
+
+        // Mock put pipeline request execution and return false
+        verify(clusterAdminClient, times(1)).putPipeline(any(PutPipelineRequest.class), actionListenerCaptor.capture());
+        actionListenerCaptor.getValue().onFailure(new Exception("Failed to create ingest pipeline"));
+
+        assertTrue(future.isDone() && future.isCompletedExceptionally());
+        try {
+            future.get();
+        } catch (ExecutionException e) {
+            assertTrue(e.getMessage().contains("Failed to create ingest pipeline"));
+        }
```

Comment:
We should test the above like the below and remove the try/catch block in the test as we want to test the exception rather catching it.
```java
assertTrue(future.isCompletedExceptionally());
ExecutionException ex = assertThrows(ExecutionException.class, () -> future.get().getContent());
assertTrue(ex.getCause() instanceof Exception);
assertEquals("Failed to create an index", ex.getCause().getMessage());
```

Reply:
Sure this looks like a better way to test this